src/Http/HttpHandler.php

```diff
@@ -59,7 +59,7 @@ public function pull(): MessageInterface
         }
 
         // Pulling response
-        preg_match('!^HTTP/(?P<version>[0-9/.]+) (?P<code>[0-9]*) (?P<reason>.*)!', $status, $matches);
+        preg_match('!^HTTP\/(?P<version>[0-9\/.]+) (?P<code>[0-9]*)(?P<reason>.*)!', $status, $matches);
```

Comment:
Those backslahes don't matter, the '!' at the start and end changes the delimiter so that they are not necessary.

What is needed it to accept "HTTP/1.1 501" with no trailing space and no reason, and I believe this does it:

`!^HTTP/(?P<version>[0-9/.]+) (?P<code>[0-9]*)($|\s(?P<reason>.*))!`

Reply:
Check, pushed a small change.
Did some testing here too using the ^ as a delimiter: https://regex101.com/r/sboq7j/1

I also notice I need to update the tests which are failing now.